src/code/z_actor.c

```diff
@@ -456,237 +469,242 @@ void Target_SetColors(TargetContext* targetCtx, Actor* actor, s32 type, PlayStat
-        y = (sp68->focus.pos.y + (sp68->targetArrowOffset * sp68->scale.y)) - targetCtx->unk0.y;
-        z = sp68->focus.pos.z - targetCtx->unk0.z;
-
-        targetCtx->unk0.x += x * temp_f0;
-        targetCtx->unk0.y += y * temp_f0;
-        targetCtx->unk0.z += z * temp_f0;
+        targetCtx->fairyHintPos.x += x * temp_f0;
+        targetCtx->fairyHintPos.y += y * temp_f0;
+        targetCtx->fairyHintPos.z += z * temp_f0;
     } else {
-        Target_SetColors(targetCtx, sp68, category, play);
+        Target_SetColors(targetCtx, actor, category, play);
     }
 
-    if ((actor != NULL) && (targetCtx->unk4B == 0)) {
-        Actor_GetProjectedPos(play, &actor->focus.pos, &projectedPos, &invW);
+    if ((targetedActor != NULL) && (targetCtx->rotZTick == 0)) {
+        Actor_GetProjectedPos(play, &targetedActor->focus.pos, &projectedPos, &invW);
         if ((projectedPos.z <= 0.0f) || (fabsf(projectedPos.x * invW) >= 1.0f) ||
             (fabsf(projectedPos.y * invW) >= 1.0f)) {
-            actor = NULL;
+            targetedActor = NULL;
         }
     }
 
-    if (actor != NULL) {
-        if (actor != targetCtx->targetedActor) {
+    if (targetedActor != NULL) {
+        if (targetedActor != targetCtx->targetedActor) {
             s32 sfxId;
 
-            func_800B4F78(targetCtx, actor->category, play);
+            // Lock On entries need to be re-initialized when changing the targeted actor
+            Target_InitLockOn(targetCtx, targetedActor->category, play);
 
-            targetCtx->targetedActor = actor;
+            targetCtx->targetedActor = targetedActor;
 
-            if (actor->id == ACTOR_EN_BOOM) {
-                targetCtx->unk48 = 0;
+            if (targetedActor->id == ACTOR_EN_BOOM) {
+                // Avoid drawing the lock on triangles on a zora boomerang
+                targetCtx->lockOnAlpha = 0;
             }
 
-            sfxId =
-                CHECK_FLAG_ALL(actor->flags, ACTOR_FLAG_4 | ACTOR_FLAG_1) ? NA_SE_SY_LOCK_ON : NA_SE_SY_LOCK_ON_HUMAN;
+            sfxId = CHECK_FLAG_ALL(targetedActor->flags, ACTOR_FLAG_TARGETABLE | ACTOR_FLAG_UNFRIENDLY)
+                        ? NA_SE_SY_LOCK_ON
+                        : NA_SE_SY_LOCK_ON_HUMAN;
             play_sound(sfxId);
         }
 
-        targetCtx->targetCenterPos.x = actor->world.pos.x;
-        targetCtx->targetCenterPos.y = actor->world.pos.y - (actor->shape.yOffset * actor->scale.y);
-        targetCtx->targetCenterPos.z = actor->world.pos.z;
+        targetCtx->targetCenterPos.x = targetedActor->world.pos.x;
+        targetCtx->targetCenterPos.y =
+            targetedActor->world.pos.y - (targetedActor->shape.yOffset * targetedActor->scale.y);
+        targetCtx->targetCenterPos.z = targetedActor->world.pos.z;
 
-        if (targetCtx->unk4B == 0) {
-            f32 temp_f0_2;
+        if (targetCtx->rotZTick == 0) {
+            f32 temp_f0_2 = (500.0f - targetCtx->lockOnRadius) * 3.0f;
             f32 clampedFloat;
```

Comment:
```suggestion
            f32 step = (500.0f - targetCtx->lockOnRadius) * 3.0f;
```
Perhaps `lockOnStep`? Also seems to be fine if you remove `clampedFloat` and just use one variable.